barretenberg/cpp/src/barretenberg/eccvm/transcript_builder.hpp

```diff
@@ -104,225 +130,397 @@ class ECCVMTranscriptBuilder {
+    }
 
+    static void add_affine_coordinates_to_transcript(std::vector<TranscriptRow>& transcript_state,
+                                                     const Accumulator& accumulator_trace,
+                                                     const Accumulator& msm_accumulator_trace,
+                                                     const Accumulator& intermediate_accumulator_trace)
+    {
         for (size_t i = 0; i < accumulator_trace.size(); ++i) {
+            TranscriptRow& row = transcript_state[i + 1];
             if (!accumulator_trace[i].is_point_at_infinity()) {
-                transcript_state[i + 1].accumulator_x = accumulator_trace[i].x;
-                transcript_state[i + 1].accumulator_y = accumulator_trace[i].y;
+                row.accumulator_x = accumulator_trace[i].x;
+                row.accumulator_y = accumulator_trace[i].y;
             }
             if (!msm_accumulator_trace[i].is_point_at_infinity()) {
-                transcript_state[i + 1].msm_output_x = msm_accumulator_trace[i].x;
-                transcript_state[i + 1].msm_output_y = msm_accumulator_trace[i].y;
+                row.msm_output_x = msm_accumulator_trace[i].x;
+                row.msm_output_y = msm_accumulator_trace[i].y;
             }
             if (!intermediate_accumulator_trace[i].is_point_at_infinity()) {
-                transcript_state[i + 1].transcript_msm_intermediate_x = intermediate_accumulator_trace[i].x;
-                transcript_state[i + 1].transcript_msm_intermediate_y = intermediate_accumulator_trace[i].y;
+                row.transcript_msm_intermediate_x = intermediate_accumulator_trace[i].x;
+                row.transcript_msm_intermediate_y = intermediate_accumulator_trace[i].y;
             }
         }
-        for (size_t i = 0; i < accumulator_trace.size(); ++i) {
-            auto& row = transcript_state[i + 1];
-            const bool msm_transition = row.msm_transition;
-            const bool add = row.q_add;
-            if (msm_transition) {
-                Element msm_output = intermediate_accumulator_trace[i];
-                row.transcript_msm_infinity = msm_output.is_point_at_infinity();
-                if (!row.transcript_msm_infinity) {
-                    transcript_msm_x_inverse_trace[i] = (msm_accumulator_trace[i].x - offset_generator().x);
-                } else {
-                    transcript_msm_x_inverse_trace[i] = 0;
-                }
-                auto lhsx = msm_output.is_point_at_infinity() ? 0 : msm_output.x;
-                auto lhsy = msm_output.is_point_at_infinity() ? 0 : msm_output.y;
-                auto rhsx = accumulator_trace[i].is_point_at_infinity() ? 0 : accumulator_trace[i].x;
-                auto rhsy = accumulator_trace[i].is_point_at_infinity() ? (0) : accumulator_trace[i].y;
-                inverse_trace_x[i] = lhsx - rhsx;
-                inverse_trace_y[i] = lhsy - rhsy;
-            } else if (add) {
```

Comment:
here add = row.q_add, but it coincides with entry.add, so the logic could be simplified a little